airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/NonDockerizedDestination.kt

```diff
@@ -70,14 +70,15 @@ class NonDockerizedDestination(
 // factory into our tests, not a pre-instantiated destination, because we want
 // to run multiple destination processes per test.
 // TODO only inject this when not running in CI, a la @Requires(notEnv = "CI_master_merge")
-@Singleton
-class NonDockerizedDestinationFactory : DestinationProcessFactory {
+// @Singleton
+class NonDockerizedDestinationFactory : DestinationProcessFactory() {
     override fun createDestinationProcess(
         command: String,
         config: ConfigurationSpecification?,
         catalog: ConfiguredAirbyteCatalog?,
         deploymentMode: TestDeploymentMode,
     ): DestinationProcess {
+        // TODO pass test name into the destination process
```

Comment:
this is nontrivial, so I'm going to just punt for now. Coroutines don't play super nicely with MDC (https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-slf4j/kotlinx.coroutines.slf4j/-m-d-c-context/)

(this is most useful for when you're running multiple tests concurrently, and most of the time if you're running locally with non-docker, you're only running one test anyway :shrug:)

naively wrapping our `destination.run()` call in a `withContext(MDCContext())` doesn't quite get there - it looks like all the tasks lose that context, so you get logs like this: (i.e. only some logs are correctly tagged with the testname/ID)

```
2024-10-08T22:10:19,481`MockBasicFunctionalityIntegrationTest.testBasicWrite()-yw9p7 > Starting input consumer task
2024-10-08T22:10:19,484`MockBasicFunctionalityIntegrationTest.testBasicWrite()-yw9p7 > Starting startup task
2024-10-08T22:10:19,484`DefaultDispatcher-worker-1 @coroutine#5`45`INFO`i.a.c.l.t.DestinationTaskScopeProvider(execute):75 - Launching task SyncTaskWrapper(innerTask=io.airbyte.cdk.load.task.internal.DefaultInputConsumerTask@e1e7b30)
2024-10-08T22:10:19,485`DefaultDispatcher-worker-3 @coroutine#6`47`INFO`i.a.c.l.t.DestinationTaskScopeProvider(execute):75 - Launching task SyncTaskWrapper(innerTask=io.airbyte.cdk.load.task.implementor.DefaultSetupTask@150888e1)
```